plugins/scaffolder/src/components/MultistepJsonForm/MultistepJsonForm.tsx

```diff
@@ -60,71 +64,37 @@ type Props = {
   fields?: FormProps<any>['fields'];
   finishButtonLabel?: string;
   layouts: LayoutOptions[];
+  ReviewStepComponent?: ComponentType<ReviewStepProps>;
 };
 
-export function getUiSchemasFromSteps(steps: Step[]): UiSchema[] {
-  const uiSchemas: Array<UiSchema> = [];
-  steps.forEach(step => {
-    const schemaProps = step.schema.properties as JsonObject;
-    for (const key in schemaProps) {
-      if (schemaProps.hasOwnProperty(key)) {
-        const uiSchema = schemaProps[key] as UiSchema;
-        uiSchema.name = key;
-        uiSchemas.push(uiSchema);
-      }
-    }
-  });
-  return uiSchemas;
-}
-
-export function getReviewData(formData: Record<string, any>, steps: Step[]) {
-  const uiSchemas = getUiSchemasFromSteps(steps);
-  const reviewData: Record<string, any> = {};
-  for (const key in formData) {
-    if (formData.hasOwnProperty(key)) {
-      const uiSchema = uiSchemas.find(us => us.name === key);
-
-      if (!uiSchema) {
-        reviewData[key] = formData[key];
-        continue;
-      }
-
-      if (uiSchema['ui:widget'] === 'password') {
-        reviewData[key] = '******';
-        continue;
-      }
-
-      if (!uiSchema['ui:backstage'] || !uiSchema['ui:backstage'].review) {
-        reviewData[key] = formData[key];
-        continue;
-      }
-
-      const review = uiSchema['ui:backstage'].review as JsonObject;
-      if (review.mask) {
-        reviewData[key] = review.mask;
-        continue;
-      }
-
-      if (!review.show) {
-        continue;
-      }
-      reviewData[key] = formData[key];
-    }
-  }
-
-  return reviewData;
+export function getSchemasFromSteps(steps: Step[]): {
+  uiSchema: UiSchema;
+  mergedSchema: JsonObject;
+  schema: JsonObject;
+}[] {
+  return steps.map(({ schema }) => ({
+    mergedSchema: schema,
+    ...extractSchemaFromStep(schema),
+  })) as unknown as {
+    uiSchema: UiSchema;
+    mergedSchema: JsonObject;
+    schema: JsonObject;
+  }[];
 }
 
-export const MultistepJsonForm = (props: Props) => {
+/**
+ * Creates the dynamic form for a scaffolder template.
+ */
+export const MultistepJsonForm = (props: MultistepJsonFormProps) => {
   const {
     formData,
     onChange,
     onReset,
     onFinish,
     fields,
     widgets,
-    finishButtonLabel,
```

Comment:
where was this used before?

Reply:
The "Create" button used to be internally configurable for label, but in `next` implementation it was just used "Create" .

Reply:
@benjdlambert 

What is in `next`:
<img width="877" alt="Screenshot 2022-11-16 at 21 25 21" src="https://user-images.githubusercontent.com/2265094/202287013-88f4de89-df2f-4ba6-8bbb-03e90910b5d9.png">

What is now:
<img width="648" alt="Screenshot 2022-11-16 at 21 26 28" src="https://user-images.githubusercontent.com/2265094/202287029-24cbfe09-6d3e-41cb-8dd9-226b5bb474a9.png">

And it also conforms with your suggestion to align with the interface https://github.com/backstage/backstage/pull/14132#discussion_r1003687544